ipaserver/plugins/certmapping.py

```diff
@@ -239,11 +248,186 @@ class cert_get_requestdata(Command):
     )
 
     def execute(self, **kw):
-        principal = kw.get('principal')
+        principal_string = kw.get('principal')
         profile_id = kw.get('profile_id', self.Backend.ra.DEFAULT_PROFILE)
         helper = kw.get('format')
 
-        result = {'debug_output': u'test'}
+        # TODO(blipton): factor this out of here and cert_request()
+        principal = split_any_principal(principal_string)
+        servicename, principal_name, realm = principal
+
+        if servicename is None:
+            principal_type = USER
+        elif servicename == 'host':
+            principal_type = HOST
+        else:
+            principal_type = SERVICE
+
+        try:
+            if principal_type == SERVICE:
+                principal_obj = api.Command.service_show(principal_string, all=True)
+            elif principal_type == HOST:
+                principal_obj = api.Command.host_show(principal_name, all=True)
+            elif principal_type == USER:
+                principal_obj = api.Command.user_show(principal_name, all=True)
+        except errors.NotFound:
+            raise errors.NotFound(
+                reason=_("The principal for this request doesn't exist."))
+        principal_obj = principal_obj['result']
+
+        request_data = self.Backend.certmapping.get_request_data(principal_obj,
+                profile_id, helper)
+
+        result = {}
+        result['debug_output'] = unicode(request_data)
         return dict(
             result=result
         )
+
+class OpenSSLHelper(object):
+    # TODO(blipton): What if we want a parenthesis in the section?
+    SECTION_RE = re.compile(r'%section\((.*?)\)', re.DOTALL)
+    EXTENSION_RE = re.compile(r'^ext:(.*)$', re.DOTALL+re.MULTILINE)
+
+    def parse_sections(self, value, sections):
```

Comment:
This should be protected or even private method, use name _process_section or __process_section